### DISCUSSION TITLE
tqdm compatible progress bar

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [zauberzeug/nicegui#1613] [Q&A] [answered]
### Question

I am trying to wrap a linear progress bar so that it shows the same behavior as a [tqdm](https://github.com/tqdm/tqdm#iterable-based#user-content-usage) progress bar. Currently i have no idea how to set the total or the description. Also the progress bar seems to fill as soon as i set a value of 1 or greater - does the progress bar expect a percentage instead?
```python
'''
Created on 2023-09-12

@author: wf
'''
from tqdm import tqdm
from nicegui import ui

class NiceguiProgressbar():
    """
    nicegui progess bar wrapper
    """

    def __init__(self,total,desc,unit):
        self.total=total
        self.value=0
        self.desc=desc
        self.unit=unit
        self.progress = ui.linear_progress(value=0).props('instant-feedback')
        self.progress.visible = False
        
    def set_description(self,desc:str):
        """
        set the description of the progress bar
        """
        self.progress.visible=True
        pass
        
    def update(self,step):
        """
        update the progress bar
        """
        self.value+=1
        self.progress.visible=True
        self.progress.value=round(self.value/self.total*100)
        pass
    
class Progressbar():
    """
    tqdm progress bar wrapper
    """
    
    def __init__(self,total,desc,unit):
        """
        constructor
        """
        self.progress=tqdm(total=total, desc=desc, unit=unit)
        
    def set_description(self,desc:str):
        self.progress.set_description(desc)
        
    def update(self,step):
        self.progress.update(step)
  ```

**Comment by rodja** [comment]
The progress is a a value between 0 and 1 (see https://nicegui.io/documentation/linear_progress). I think we need something similar to https://github.com/zauberzeug/nicegui/pull/1493 to put ui.label or similar "into" the progress bar.

**Reply by falkoschindler** [reply]
https://github.com/zauberzeug/nicegui/pull/1493 already fixed circular _and_ linear progress bars.

**Comment by falkoschindler** [comment, marked as answer]
Here is an example how you can place a custom label inside a linear progress bar:
```py
slider = ui.slider(min=0, max=1, step=0.001, value=0.5)
with ui.linear_progress(show_value=False, size='2em').props('instant-feedback').bind_value_from(slider, 'value'):
    label = ui.label('test') \
        .classes('text-lg text-black absolute-center') \
        .bind_text_from(slider, 'value', backward=lambda v: f'Making progress... {v:.1%}')
```

<img width="488" alt="Screenshot 2023-09-15 at 19 18 51" src="https://github.com/zauberzeug/nicegui/assets/5767091/90c66438-7609-4c69-bd05-47f66e587020">

**Comment by WolfgangFahl** [comment]
@falkoschindler Thanks for the suggestion. I have started implementing this - there is just follow up issues with the value range and the overlap
![grafik](https://github.com/zauberzeug/nicegui/assets/1336221/f9f40e3b-f6d4-4452-b1d9-48eab9844809)
